WooCommerce/Classes/Analytics/WooAnalytics.swift

```diff
@@ -5,22 +5,26 @@ import WordPressShared
 import WidgetKit
 import enum Alamofire.AFError
 import Yosemite
+import enum WooFoundation.WooAnalyticsStat
+import protocol WooFoundation.Analytics
+import protocol WooFoundation.AnalyticsProvider
+import struct WooFoundation.WooAnalyticsEvent
```

Comment:
The type import here make me wonder if `WooAnalyticsStat` and `WooAnalyticsEvent` should be in WooFoundation, or be kept in WooCommerce now that we've moved the protocols over:

While `WooAnalyticsEvent` is nice for consistency for event tracking, it feels that `WooAnalyticsStat` now becomes an implementation detail of the WooCommerce app, and since what we want to track in the POS framework (and potentially separate app) would most likely be be different events, `WooAnalyticsStat` perhaps should stay in WooCommerce. Thoughts?

Reply:
Yup I had similar thoughts before and these aren't necessary indeed, I shared the reasons in the description with the following highlights - 

- `WooAnalyticsStat` enum: It's nice having a consolidated list of events to avoid accidentally having two events of the same name
- `WooAnalyticsEvent` struct: this one is an extension of the protocol and isn't necessary, since this struct is to allow type-checking on the event properties and a cleaner way of specifying events. But I thought this could make event tracking code easier in the POS framework as well

WDYT + @joshheald @bozidarsevo? I'm happy to make any changes after we make a decision.

Reply:
>  It's nice having a consolidated list of events to avoid accidentally having two events of the same name.

This makes sense, but at the same time, the existing events are (as @iamgabrielma says) closely coupled to the existing app. We're likely to _want_ to have two events with the same name, e.g. `collect_payment_tapped`

Perhaps we should discuss with the wider team the idea of having a different prefix for POS events instead, i.e. `woocommerceposios_collect_payment_tapped` to namespace these separately. If we did that, we can leave the enum and struct in the existing app and just extract a protocol for `WooAnalyticsEvent` to put in `WooFoundation`.

We should do the same thing on both platforms though.

Reply:
Also... if we do need to import 4 things from a framework, it feels to me like it's time to start importing the framework. Just my 2¢, but individual lines for each file adds clutter without value.

Reply:
Thanks for the feedback @iamgabrielma @joshheald! It really helped clear up my original doubts and debates. I've moved the event enum and struct back to the main app in https://github.com/woocommerce/woocommerce-ios/pull/12685/commits/264b8b48666682a122f63880030661a8165a4a71 with a bunch of other updates and polishes in the latest commits.

> Perhaps we should discuss with the wider team the idea of having a different prefix for POS events instead, i.e. `woocommerceposios_collect_payment_tapped` to namespace these separately. If we did that, we can leave the enum and struct in the existing app and just extract a protocol for `WooAnalyticsEvent` to put in `WooFoundation`.
> 
> We should do the same thing on both platforms though.

If we have plans to move POS to a separate app, personally I think keeping the `woocommerceios_` prefix with the additional POS mention like `woocommerceios_pos_` would be easier to differentiate events between the 2 apps. In any case, for the context in this PR, no decision is made in the POS framework and we can defer the POS implementation later.